corehq/util/django_migrations.py

```diff
@@ -53,9 +53,13 @@ def add_if_not_exists_raw(string, name):
 
 class DatabaseSchemaEditorIfNotExists(DatabaseSchemaEditor):
     sql_create_index = add_if_not_exists(DatabaseSchemaEditor.sql_create_index)
-    sql_create_varchar_index = add_if_not_exists(DatabaseSchemaEditor.sql_create_varchar_index)
-    sql_create_text_index = add_if_not_exists(DatabaseSchemaEditor.sql_create_text_index)
     sql_create_unique = add_if_not_exists(DatabaseSchemaEditor.sql_create_unique)
+    # can remove the following lines once we're on Django 2.x
+    # they're internals whose usage within Django was replaced
+    if hasattr(DatabaseSchemaEditor, 'sql_create_varchar_index'):
+        sql_create_varchar_index = add_if_not_exists(DatabaseSchemaEditor.sql_create_varchar_index)
+    if hasattr(DatabaseSchemaEditor, 'sql_create_text_index'):
+        sql_create_text_index = add_if_not_exists(DatabaseSchemaEditor.sql_create_text_index)
```

Comment:
will these still work without them?

Reply:
I had a hunch that they would, because these are internals that were removed and whose calls were replaced within `django`. My hunch is that the calls would be routed through the other properties this overrides.